resources/js/components/LinkableTrendMetric.vue

```diff
@@ -1,177 +1,149 @@
-      handleRangeSelected(key) {
-        this.selectedRangeKey = key
-        this.fetch()
-      },
-
-      fetch() {
-        this.loading = true
-
-        minimum(Nova.request().get(this.metricEndpoint, this.metricPayload)).then(
-          ({
-            data: {
-              value: {
-                labels,
-                trend,
-                value,
-                prefix,
-                suffix,
-                suffixInflection,
-                format,
-              },
-            },
-          }) => {
-            this.value = value
-            this.labels = Object.keys(trend)
-            this.data = {
-              labels: Object.keys(trend),
-              series: [
-                map(trend, (value, label) => {
-                  return {
-                    meta: label,
-                    value: value,
-                  }
-                }),
-              ],
-            }
-            this.format = format || this.format
-            this.prefix = prefix || this.prefix
-            this.suffix = suffix || this.suffix
-            this.suffixInflection = suffixInflection
-            this.loading = false
-          }
-        )
-      },
+    hasRanges() {
+      return this.card.ranges.length > 0
     },
 
-    computed: {
-      hasRanges() {
-        return this.card.ranges.length > 0
-      },
-
-      metricPayload() {
-        const payload = {
-          params: {
-            timezone: this.userTimezone,
-            twelveHourTime: this.usesTwelveHourTime,
-          },
-        }
+    metricPayload() {
+      const payload = {
+        params: {
+          timezone: this.userTimezone,
+          twelveHourTime: this.usesTwelveHourTime,
+        },
+      }
 
-        if (
-          !Nova.missingResource(this.resourceName) &&
-          this.card &&
-          this.card.refreshWhenFiltersChange === true
-        ) {
-          payload.params.filter =
-            this.$store.getters[`${this.resourceName}/currentEncodedFilters`]
-        }
+      if (
+        !Nova.missingResource(this.resourceName) &&
+        this.card &&
+        this.card.refreshWhenFiltersChange === true
+      ) {
+        payload.params.filter =
+          this.$store.getters[`${this.resourceName}/currentEncodedFilters`]
+      }
 
-        if (this.hasRanges) {
-          payload.params.range = this.selectedRangeKey
-        }
+      if (this.hasRanges) {
+        payload.params.range = this.selectedRangeKey
+      }
 
-        return payload
-      },
-
-      metricEndpoint() {
-        const lens = this.lens !== '' ? `/lens/${this.lens}` : ''
-        if (this.resourceName && this.resourceId) {
-          return `/nova-api/${this.resourceName}${lens}/${this.resourceId}/metrics/${this.card.uriKey}`
-        } else if (this.resourceName) {
-          return `/nova-api/${this.resourceName}${lens}/metrics/${this.card.uriKey}`
-        } else {
-          return `/nova-api/metrics/${this.card.uriKey}`
-        }
-      },
```

Comment:
3rd party component no longer need to declare computed `metricEndpoint` as this is always available via `MetricBehaviour` mixin